### DISCUSSION TITLE
Barman Backup Using all Disk Space During Backup Process

### DISCUSSION THREAD

**Discussion by xangxiong** [cloudnative-pg/cloudnative-pg#3758] [Q&A] [answered]
Most of my K8S nodes have 100GB disk available for POD overlay volume.  As a result, all Postgres PODs only have around 100GB of space to use for storing the backup part files generated by Barman.

The Barman Backup command would generate the part files in `/controller/backup/barman-cloud-backup-XXXX`.  Since `/controller` is mounted as part of the `/` volume, during backup, the entire 100GB is not enough.

The barman-cloud-backup command supports the option `max-archive-size` which defaults to 100GB.

I'm wondering if there is a workaround for this limitation.   Are there possibly:

1. Ways to override/specify the `max-archive-size`?
2. Specify a volume for `/controller/backup` so that we can easily increase the volume size without having to increase the disk size of each Kubernetes Node?

I'm open to any suggestions or help for this.  I'm leaving increasing each Kubernetes node boot disk size as a last option.

**Comment by xangxiong** [comment, marked as answer]
Was able to find a workaround using a custom build postgresql image with an updated /usr/loca/bin/barman-cloud-backup script. 

The updated script is:

```
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import re
import os
import sys
from barman.clients.cloud_backup import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    
    has_max_archive_size = False
    for arg in sys.argv:
        if '--max-archive-size' in arg:
            has_max_archive_size = True
            break

    if not has_max_archive_size:
        # we will try and use the value from Env if it exists
        if os.environ.get('MAX_ARCHIVE_SIZE', None):
            sys.argv.append('--max-archive-size')
            sys.argv.append(os.environ['MAX_ARCHIVE_SIZE'])
        
    sys.exit(main())
```

With the new postgresql image, I updated my deployment to:

```
apiVersion: postgresql.cnpg.io/v1
kind: Cluster
metadata:
  name: test
  namespace: default
spec:
  instances: 3
  imageName: mycustom/postgresql:16.1
  env:
    - name: MAX_ARCHIVE_SIZE
      value: 20GB
...
```

**Comment by markmaker** [comment]
You should be able to set the `--max-archive-size` via cluster manifest, right? If true then no custom build is needed.

https://cloudnative-pg.io/plugin-barman-cloud/docs/misc/#example-extra-backup-options

**Comment by dmmorano** [comment]
Do the official helm charts not have wiring for this?
